File: src/main/java/mediathek/gui/abo/ManageAboDialog.java
package mediathek.gui.abo;

import javafx.application.Platform;
import javafx.embed.swing.JFXPanel;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import mediathek.config.Daten;
import mediathek.config.MVConfig;
import mediathek.tool.EscapeKeyHandler;
import mediathek.tool.GuiFunktionen;

import javax.swing.*;
import java.awt.*;
import java.net.URL;

public class ManageAboDialog extends JDialog {
    private final ManageAboPanel aboPanel;
    private final JFXPanel infoPanel = new JFXPanel();
    private Parent infoPane;
    private AboInformationController infoController;

    public ManageAboDialog(Frame owner, Daten daten) {
        super(owner);
        setTitle("Abos verwalten");
        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
        setResizable(true);

        aboPanel = new ManageAboPanel(daten);
        var contentPane = getContentPane();
        contentPane.setLayout(new BorderLayout());
        contentPane.add(aboPanel, BorderLayout.CENTER);
        contentPane.add(infoPanel, BorderLayout.SOUTH);

        setupControllers();

        pack();
        //restore saved size
        GuiFunktionen.setSize(MVConfig.Configs.SYSTEM_GROESSE_MANAGE_ABO, this, owner);

        EscapeKeyHandler.installHandler(this, this::dispose);
    }

    private void setupControllers() {
        Platform.runLater(() -> {
            try {
                URL url = getClass().getResource("/mediathek/res/programm/fxml/abo_information_panel.fxml");

                FXMLLoader loader = new FXMLLoader();
                loader.setLocation(url);

                infoPane = loader.load();
                /*
                 * controller must be kept in variable for strong ref, otherwise GC will erase controller and therefore
                 * update of abos in dialog will stop working...
                 */
                infoController = loader.getController();

                Scene scene = new Scene(infoPane);
                infoPanel.setScene(scene);
            } catch (Exception ex) {
                ex.printStackTrace();
            }
        });
    }

    public ManageAboPanel getAboPanel() {
        return aboPanel;
    }
}

File: src/main/java/mediathek/gui/abo/ManageAboDialog.kt
package mediathek.gui.abo

import mediathek.tool.ApplicationConfiguration
import mediathek.tool.EscapeKeyHandler
import org.apache.commons.configuration2.sync.LockMode
import java.awt.BorderLayout
import java.awt.Frame
import javax.swing.JDialog

class ManageAboDialog(owner: Frame?) : JDialog(owner) {
    private val aboPanel: ManageAboPanel

    override fun dispose() {
        aboPanel.tabelleSpeichern()
        saveToConfig()

        super.dispose()
    }

    companion object {
        private const val HEIGHT = "manage_abo_dialog.height"
        private const val WIDTH = "manage_abo_dialog.width"
        private const val X = "manage_abo_dialog.x"
        private const val Y = "manage_abo_dialog.y"
    }

    private fun restoreFromConfig() {
        val config = ApplicationConfiguration.getConfiguration()
        try {
            config.lock(LockMode.READ)
            val height = config.getInt(HEIGHT)
            val width = config.getInt(WIDTH)
            val x = config.getInt(X)
            val y = config.getInt(Y)

            setSize(width, height)
            setLocation(x,y)
        }
        catch(_: NoSuchElementException) {
        }
        finally {
            config.unlock(LockMode.READ)
        }
    }

    private fun saveToConfig() {
        val config = ApplicationConfiguration.getConfiguration()
        try {
            config.lock(LockMode.WRITE)
            config.setProperty(HEIGHT, size.height)
            config.setProperty(WIDTH, size.width)
            config.setProperty(X, location.x)
            config.setProperty(Y, location.y)
        }
        finally {
            config.unlock(LockMode.WRITE)
        }
    }

    init {
        title = "Abos verwalten"
        defaultCloseOperation = DISPOSE_ON_CLOSE
        isResizable = true
        isModal = true
        aboPanel = ManageAboPanel()
        val contentPane = contentPane
        contentPane.layout = BorderLayout()
        contentPane.add(aboPanel, BorderLayout.CENTER)
        pack()

        restoreFromConfig()

        EscapeKeyHandler.installHandler(this) { dispose() }
    }
}